analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/NullPointerDereferenceBase.cs

```diff
@@ -31,16 +31,17 @@ public abstract class NullPointerDereferenceBase : SymbolicRuleCheck
 
         protected override ProgramState PreProcessSimple(SymbolicContext context)
         {
-            switch (context.Operation.Instance.Kind)
+            var operation = context.Operation.Instance switch
```

Comment:
Let's use switch expression, but with switching on the Kind. Delegating kind to the pattern matching increases complexity.